src/plugins/blocksmith/components/instagram-media-block-preview/index.jsx

```diff
@@ -1,17 +1,84 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Message from '@gdbots/pbj/Message';
-import InstagramEmbed from 'react-instagram-embed';
+import './styles.scss';
 
-const InstagramMediaBlockPreview = ({ block }) => (
-  <InstagramEmbed
-    url={`https://instagr.am/p/${block.get('id')}/`}
-    hideCaption={block.get('hidecaption') || false}
-  />
-);
+class InstagramPostBlockPreview extends Component {
+  constructor(props) {
+    super(props);
+    this.embedParentRef = React.createRef();
+    this.clean = this.clean.bind(this);
+    this.embed = this.embed.bind(this);
+  }
 
-InstagramMediaBlockPreview.propTypes = {
+  componentDidMount() {
+    this.addInstagramEmbedScript();
+    this.embed();
+  }
+
+  componentDidUpdate() {
```

Comment:
@tbone1020 you only have to do the stuff in here if the block's id or hidecaption changes. if you check for that then you can prevent unnecessary renders when someone is changing "is update" or "aside"